libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2213,7 +2213,11 @@ void TypeChecker::typeCheckBytesConcatFunction(
 )
 {
```

Comment:
Yeah - in here you now need to adjust the type-checking of the arguments for ``string.concat``.
There actually may be little in common here between ``bytes.concat`` and ``string.concat``. ``string.concat`` should only allow arguments that are implicitly convertible to ``string memory```. You can case-distinguish in this function or it may even call for adding a new function ``typeCheckStringConcatFunction`` to do that.